oauth2_http/java/com/google/auth/oauth2/LoggingUtils.java

```diff
@@ -0,0 +1,155 @@
+        loggingDataMap.put("request.method", request.getRequestMethod());
+        loggingDataMap.put("request.url", request.getUrl().toString());
+
+        Map<String, Object> headers = new HashMap<>();
+        request
+            .getHeaders()
+            .forEach(
+                (key, val) -> {
+                  if ("authorization".equals(key)) {
+                    String hashedVal = calculateSHA256Hash(String.valueOf(val));
+                    headers.put(key, hashedVal);
+                  } else {
+                    headers.put(key, val);
+                  }
+                });
+        loggingDataMap.put("request.headers", headers.toString());
+
+        if (request.getContent() != null && logger.isDebugEnabled()) {
+          // are payload always GenericData? If so, can parse and store in json
+          GenericData data = (GenericData) ((UrlEncodedContent) request.getContent()).getData();
+
+          Map<String, String> contextMap = parseGenericData(data);
+          loggingDataMap.put("request.payload", contextMap.toString());
+
+          logWithMDC(logger, org.slf4j.event.Level.DEBUG, loggingDataMap, message);
+        } else {
+
+          logWithMDC(logger, org.slf4j.event.Level.INFO, loggingDataMap, message);
+        }
+      } catch (Exception e) {
+        logger.error("Error logging request: ", e);
+      }
+    }
+  }
+
+  static void logResponse(HttpResponse response, Logger logger, String message) {
+    if (logger.isInfoEnabled()) {
+      try {
+        Map<String, String> responseLogDataMap = new HashMap<>();
+        responseLogDataMap.put("response.status", String.valueOf(response.getStatusCode()));
+        responseLogDataMap.put("response.status.message", response.getStatusMessage());
+
+        Map<String, Object> headers = new HashMap<>();
+        response.getHeaders().forEach((key, val) -> headers.put(key, val));
+        responseLogDataMap.put("response.headers", headers.toString());
+        logWithMDC(logger, org.slf4j.event.Level.INFO, responseLogDataMap, message);
+      } catch (Exception e) {
+
+        logger.error("Error logging response: ", e);
+      }
+    }
+  }
+
+  static void logGenericData(GenericData genericData, Logger logger, String message) {
+    if (logger.isDebugEnabled()) {
+      try {
+
+        Map<String, String> contextMap = parseGenericData(genericData);
+        logWithMDC(logger, org.slf4j.event.Level.DEBUG, contextMap, message);
+      } catch (Exception e) {
+        logger.error("Error logging GenericData: ", e);
+      }
+    }
+  }
+
+  private static Map<String, String> parseGenericData(GenericData genericData) {
+    Map<String, String> contextMap = new HashMap<>();
+    genericData.forEach(
+        (key, val) -> {
+          if ("token".equals(key)
+              || "assertion".equals(key)
+              || "access_token".equals(key)
+              || "client_secret".equals(key)
+              || "refresh_token".equals(key)) {
```

Comment:
Perhaps we can have these in a list + authorization so that we can do something like `if (list.contains(key))`, then hash the value?